### PR TITLE
use abstract method to define static oracle callback

### PR DESCRIPTION
**Pull request by Jim8y**
This pr updates the IOracle to define the oracle callback interface as static https://github.com/neo-project/neo-devpack-dotnet/pull/993#discussion_r1529952710

### CODE REVIEW

**Review by cschuchardt88 — APPROVED** [review]

**Comment by shargon** [comment]
But why we want to use static interface in oracle, and not in others? we should use the same way

File: tests/Neo.SmartContract.Framework.UnitTests/TestingArtifacts/Contract_IOracle.cs
Comment (cschuchardt88) [inline tests/Neo.SmartContract.Framework.UnitTests/TestingArtifacts/Contract_IOracle.cs:1]: I thought we were using `block namespace` scoping?

**Comment by Jim8y** [comment]
@shargon i am sorry to say this but i think it is a mess at all cause static or non-static all works as interface..... People can use it as static or non-static... how do we make an interace that both works for static and non-static @cschuchardt88

**Comment by shargon** [comment]
> @shargon i am sorry to say this but i think it is a mess at all cause static or non-static all works as interface..... People can use it as static or non-static... how do we make an interace that both works for static and non-static @cschuchardt88 

We should try to focus on instance

**Comment by Jim8y** [comment]
Use instance interface instead.

**Pull request closed without merging**